File: reactive-java/src/main/java/com/example/reactivejava/service/EmployeeService.java
package com.example.reactivejava.service;

import com.example.reactivejava.model.Employee;
import com.example.reactivejava.repository.EmployeeRepo;
import reactor.core.publisher.Mono;

import org.springframework.stereotype.Service;

@Service
public class EmployeeService {

	private final EmployeeRepo employeeRepo;

	EmployeeService(EmployeeRepo employeeRepo) {
		this.employeeRepo = employeeRepo;
	}

	public Mono<Employee> getData(String id) {
		return employeeRepo.findEmployeeById(id);
	}
}

File: reactive-kotlin/src/main/kotlin/com/example/reactivekotlin/service/EmployeeService.kt
package com.example.reactivekotlin.service

import com.example.reactivekotlin.model.Employee
import com.example.reactivekotlin.repository.EmployeeRepo
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class EmployeeService(
    private val employeeRepo: EmployeeRepo
) {
    fun getData(id: String?): Mono<Employee> {
        return employeeRepo.findEmployeeById(id)
    }
}